File: 1Week6-Python/sentimental-readability/readability.py
# TODO

text = input("Text: ")

letters = 0
words = 0
fullsentences = 0

for char in text:
    if char.isalpha():
        letters += 1

for char in text:
    if char.isspace():
        words += 1

words += 1

for char in text:
    if char in ['!', '?', '.']:
        fullsentences += 1

L = (letters / words) * 100
S = (fullsentences / words) * 100

index = round(0.0588 * L - 0.296 * S - 15.8)

if index < 1:
    print("Before Grade 1")
elif (index > 16):
    print("Grade 16+")
else:
    print(f"Grade {index}")

File: readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <ctype.h>
#include <math.h>

int main(void)
{
    int letters = 0;
    int words = 0;
    int fullsentences = 0;

    string sentence = get_string("Text: ");

    for (int i = 0; sentence[i] != 0; i++)
    {
        if (isalpha(sentence[i]))
        {
            letters++;
        }
    }

    for (int j = 0; sentence[j] != 0; j++)
    {
        if (sentence[j] == ' ')
        {
            words++;
        }
    }

    for (int k = 0; sentence[k] != 0; k++)
    {
        if (sentence[k] == '!' || sentence[k] == '?' || sentence[k] == '.')
        {
            fullsentences++;
        }
    }

    words++;

    float L = (letters / (float) words) * 100;

    float S = (fullsentences / (float) words) * 100;

    int index = round(0.0588 * L - 0.296 * S - 15.8);


    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index > 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", index);
    }

}